app/lib/screens/history/history.dart

```diff
@@ -11,6 +18,36 @@ class _TravelHistoryState extends State<TravelHistory> {
     print('Init: Travel history');
   }
 
+  final pdf = pw.Document(); //Create a blank document
+  //add data into the pdf file
+  buildPdf() {
+    pdf.addPage(pw.MultiPage(
+      pageFormat: PdfPageFormat.a4,
+      margin: pw.EdgeInsets.all(34),
+      build: (pw.Context context) {
+        return <pw.Widget>[
+          /*
+              We will be calling API and fetching data from the database here.
+              Now just adding random text for experimentation purpose. Once we have databse/API set up, will
+              experimenting on fetching data
+              */
+          pw.Header(level: 0, child: pw.Text("Shipper Information")),
+          pw.Paragraph(text: 'Some Information'),
+          pw.Paragraph(text: 'Some Information'),
+          pw.Header(level: 1, child: pw.Text("Travel Information")),
+          pw.Paragraph(text: 'Some Text 1'),
+          pw.Paragraph(text: 'Some Text 2'),
+        ];
+      },
+    ));
+  }
+
+  Future savePdf() async {
+    final output = await getTemporaryDirectory();
+    final file = File("${output.path}/example.pdf");
+    file.writeAsBytesSync(await pdf.save());
```

Comment:
I think this applies to any async/await calls, it's best to wrap this with try/catch block to handle exception errors

Reply:
Not sure if related to this comment or [this one](https://github.com/holtzmak/Humane-Transport/pull/110/files?file-filters%5B%5D=.dart&file-filters%5B%5D=.lock#r556983781), but I had the simulator open with your PDF page open, and my machine started to sound like it was about to launch into outer space.

- [ ] Check that generating this PDF doesn't also do this to your machine. Running the application with Clark's recent PR did not jump my machine into hyperspeed

Reply:
> I think this applies to any async/await calls, it's best to wrap this with try/catch block to handle exception errors

Yes, during my research I also found that if you want to have exception handling in async function we can defiantly do that. I have added exception handling with async/await.